piksi_status_indicator/python/piksi_status_indicator/base_station_indicator.py

```diff
@@ -0,0 +1,141 @@
+
+
+class BaseStationIndicator(object):
+    """Base Station Indicator using the Neostate package.
+
+    During survey the leds will increasingly turn green until the sampling has finished.
+    Once finished the leds will stay green for 3 seconds and then use blue indicators
+    to show how many satellites the base station is seeing (more than 35 satellites results
+    in all leds turned on).
+    If the blue indicators are blinking, then the base station fixed mode is not in SBAS. 
+    """
+    def __init__(self):
+        rospy.init_node("piksi_status_indicator")
+
+        while not rospy.has_param("piksi_multi_cpp_base/num_desired_fixes"):
+            rospy.logwarn_throttle(5, "[Piksi Status Indicator]: Waiting for <num_desired_fixes> param")
+        self.desired_fixes_ = rospy.get_param("piksi_multi_cpp_base/num_desired_fixes")
+
+        num_leds = rospy.get_param("~number_of_leds")
+        self.indicator = LEDArray(num_leds)
+        
+        self.finished_sampling = False
+        self.finished_survey = False
+
+        self.green_color_ = [0, 155, 0] 
+        self.blue_color_ = [0, 0, 155] 
+        
+        #TODO(clanegge): Search for namespace of base station
+        self.kf_pos_sub_ = rospy.Subscriber(
+            "/piksi_multi_cpp_base/base_station_receiver_0/position_sampler/kf_position",
+            PositionWithCovarianceStamped,
+            self.set_sampling_indicators,
+            queue_size=1,
+        )
+
+        self.recv_state_sub_ = rospy.Subscriber(
+            "piksi_multi_cpp_base/base_station_receiver_0/ros/receiver_state",
+            ReceiverState_V2_6_5,
+            self.set_satellite_indicators,
+            queue_size=1,
+        )
+
+        # Turn off all leds to indicate script is running
+        self.indicator.turn_off_all_leds()
+
+
+    def run(self):
+        rospy.loginfo("[Piksi Status Indicator]: Starting base station LED indicator..")
+        rospy.spin()
+    
+    def get_next_off_led(self):
+        return  next((idx for idx, val in enumerate(self.indicator.led_status) if val is 0), None)
+
+    def set_sampling_indicators(self, msg):
+        if msg.header.seq == (self.desired_fixes_ -1):
+            # Sampling finished 
+            self.finished_sampling = True
+            self.ml_pos_sub_ = rospy.Subscriber(
+            "/piksi_multi_cpp_base/base_station_receiver_0/position_sampler/ml_position",
+            PositionWithCovarianceStamped,
+            self.set_finish_survey_indicators,
+            queue_size=1,
+            )
+
+        if not self.finished_sampling:
+            # look for the first not blinking led:
+            next_id = self.get_next_off_led()
+            # Split leds depending on total desired fixes on how many leds are available
+            # as long as there are still some leds which are off
+            if (next_id is not None and
+                msg.header.seq
+                >= self.desired_fixes_ / self.indicator.get_num_leds() * next_id
```

Comment:
I see, you use the `desired_fixes_` also to indicate the progress of the sampling. I think we should not only publish the current position estimate in the sampling process but also the progress of the sampling in a future PR.